### PR TITLE
(DOCS-9683) Add Azure resource collection

### PR DESCRIPTION
**Pull request by buraizu**
### What does this PR do? What is the motivation?
Add information for setting up Azure resource change collection

### Merge instructions

Merge readiness:
- [ ] Ready for merge

Merge queue is enabled in this repo. To have it automatically merged after it receives the required reviews, create the PR (from a branch that follows the `<yourname>/description` naming convention) and then add the following PR comment:

```
/merge
```

### CODE REVIEW

File: content/en/infrastructure/resource_catalog/_index.md [hunk 1]
```diff
@@ -142,6 +142,10 @@ Use the **Recent Changes** tab on a resource's side panel to see a 7-day history
 
 See the [AWS Config integration page][6] to launch a CloudFormation template that sets up change event forwarding through AWS Config.
 
+#### Azure
+
+After confirming that you have enabled resource collection under the **Resource Collection** tab in the [Azure integration tile][14], submit your request using the [form above](#recent-changes). There are no other setup steps.
```
Comment (sriramkraman) [inline content/en/infrastructure/resource_catalog/_index.md:147]: @andresromerodatadog Does the wording here look ok? 

@buraizu Maybe recommend switching to something like this - 
"After you've submitted your request to the form above, please ensure Resource Collection is turned on for the Azure subscriptions in the [Azure Integration tile][14] to also collect resource configuration changes."

**Review by sriramkraman — APPROVED** [review]

File: content/en/infrastructure/resource_catalog/_index.md [hunk 2]
```diff
@@ -142,6 +142,10 @@ Use the **Recent Changes** tab on a resource's side panel to see a 7-day history
 
 See the [AWS Config integration page][6] to launch a CloudFormation template that sets up change event forwarding through AWS Config.
 
+#### Azure
+
+After you've submitted your request to the form above, ensure **Resource Collection** is enabled for the Azure subscriptions in the [Azure integration tile][14] to also collect resource configuration changes.
```
Comment (brett0000FF) [inline content/en/infrastructure/resource_catalog/_index.md:147]:
```suggestion
To collect resource configuration changes, enable **Resource Collection** for your Azure subscriptions in the [Azure integration tile][14].
```

**Review by brett0000FF — APPROVED** [review]
Approved with one small suggestion.

**Pull request merged**